src/ResultWriter/WaveFieldWriter.cpp

```diff
@@ -252,28 +260,32 @@ void seissol::writer::WaveFieldWriter::init(unsigned int numVars,
 
 		subVertices.resize(oldToNewVertexMap.size());
 
-		// Loop over the map and assign the vertices
-		for (std::map<int,int>::iterator it=oldToNewVertexMap.begin(); it!=oldToNewVertexMap.end(); ++it)
-			subVertices.at(it->second) = &allVertices[it->first];
+		for (auto & it : oldToNewVertexMap) {
+            subVertices.at(it.second) = &allVertices[it.first];
+		}
 
 		numElems = subElements.size();
 		numVerts = subVertices.size();
 
 		meshRefiner = new refinement::MeshRefiner<double>(subElements, subVertices,
 			oldToNewVertexMap, *tetRefiner);
-	} else {
-		meshRefiner = new refinement::MeshRefiner<double>(meshReader, *tetRefiner);
+    } else {
+        meshRefiner = new refinement::MeshRefiner<double>(meshReader, *tetRefiner);
 		m_map = map;
 	}
 
+    if (numElems == 0) {
+        logError() << "All cells are excluded in the WaveFieldWriter.";
```

Comment:
WaveFieldWriter: All cells have been filtered out (OutputRegionBounds and OutputGroups).